lib/rules/prep-for-comparison.js

```diff
@@ -0,0 +1,29 @@
+const isString = require('lodash.isstring');
+const isBoolean = require('lodash.isboolean');
+const isNumber = require('lodash.isnumber');
+const isObject = require('lodash.isobject');
+
+function prepForComparison(value, rule, recursive = false) {
+  let coercedValue = value;
+  // coerce boolean/number/string to string, lowercase if case insensitive rule
+  if (isBoolean(coercedValue) || isNumber(coercedValue) || isString(coercedValue)) {
+    coercedValue = String(coercedValue);
+    if (!rule.isCaseSensitive) {
+      coercedValue = coercedValue.toLowerCase();
+    }
+  } else if (!recursive && !rule.isCaseSensitive) {
+    // if rule is case insensitive and we're not in a recursive call
+    // then prep array/object values for comparison
+    if (Array.isArray(coercedValue)) {
+      coercedValue = coercedValue.map(val => prepForComparison(val, rule, true));
+    } else if (isObject(coercedValue)) {
+      coercedValue = Object.keys(coercedValue).reduce((memo, key) => ({
```

Comment:
Since `lodash.includes` doesn't support deep includes, only shallow, seems like we save time from a deep recursion by extracting out the actual coercion into a separate function, `coerce`,  and just call that for each shallow property value. Same approach for the `Array`.

Reply:
we are only doing shallow coercion here.  the recursive call only modifies boolean/number/string.